### ISSUE TITLE
Use thirdparty GNU style flag packages

### ISSUE THREAD

**Issue by yegle** [aisola/go-coreutils#42]
The `flag` package in golang doesn't support combination of short flags.

e.g. in GNU style, `-help` means `-h -e -l -p`, and `-n1` means `-n=1`.

But in Go `-help` means `--help`, and `-n1` means `--n1`. Without a proper GNU style flag parser, there's no easy way to support things like `tail -n10`.

Here's a short list of packages I googled that do GNU style flag parsing:

* https://github.com/ogier/pflag
* https://launchpad.net/gnuflag
* https://github.com/alecthomas/kingpin

I'll be happy to provide some PR if any of the packages are accepted as replacement.

**Comment by yegle** [comment]
Also thirdparty package candidate: https://code.google.com/p/getopt/

**Comment by mattn** [comment]
+1

**Comment by mattn** [comment]
I do always

```
$ ls -la
$ rm -rf ...
```

etc...

**Comment by yegle** [comment]
I've played and tested https://github.com/ogier/pflag This is a good drop-in replacement to Go's `flag` package (e.g. just change the import path and everything works fine). But it lacks some feature if we want to use it to simulate all `coreutils` flag behaviour.

One notable things that `pflag` can't do is the optional argument for non-bool flags. The example is `ls --color` which means `ls --color=always`. This is a `flag.String` in `pflag`, but you'll have to supply a value for such a flag.

I've reported this issue https://github.com/ogier/pflag/issues/16 but I don't think there's an easy fix for the problem.

Moving on to test the other packages.
